### ISSUE TITLE
Remove empty errors in Graphql response

### ISSUE THREAD

**Issue by mingfang** [semi-technologies/weaviate#1535]
add `omitempty` to 
https://github.com/semi-technologies/weaviate/blob/a94ebaed51536951cf063b9af69ced4709c686a0/entities/models/graph_q_l_response.go#L36

Like this
```go
Errors []*GraphQLError `json:"errors,omitempty"`
```

Currently a query without any errors would still contain an empty error in the response.
```json
{
  "data": {
    "Get": {
      "Superhero": [
        {
          "name": "Batman"
        },
        {
          "name": "Superman"
        }
      ]
    }
  },
  "errors": null
}
```

The expected response is without any error at all
```json
{
  "data": {
    "Get": {
      "Superhero": [
        {
          "name": "Batman"
        },
        {
          "name": "Superman"
        }
      ]
    }
  }
}
```

Currently this breaks Hasura when adding Weaviate as a remote schema with this error.
```json
{
  "errors": [
    {
      "extensions": {
        "path": "$",
        "code": "remote-schema-error"
      },
      "message": "Received invalid JSON value from remote"
    }
  ]
}
```

**Comment by etiennedi** [comment]
Hi @mingfang -

Thanks for discovering this and the detailed bug report. We're going to look into it. It might not be as simple as setting the `omitempty` flag, becaues the file you referenced is being auto-generated by `go-swagger`. I hope we can set an option there that leads to an `omitempty` flag.

**Comment by etiennedi** [comment]
In case anyone ever stumbles upon this issue in the future and thinks: What do the GraphQL specs say about this? Turns out [they're very explicit](https://spec.graphql.org/June2018/#sec-Errors):

> If no errors were encountered during the requested operation, the errors entry should not be present in the result.

Going to fix this.

**Comment by etiennedi** [comment]
Hi @mingfang, the issue has been fixed and will be included in the next public release.

A pre-release docker image which contains the fix is already available as `semitechnologies/weaviate:1.2.1-f1bfc5e`. Feel free to use this image to verify if the issue is gone for your case. If not, please reopen this issue. Thanks.
